### PR TITLE
Add region fixer and refactor the write_log

### PR DESCRIPTION
**Pull request by Lazr1026**
* The code will check for an installed Wii U Menu and set the region values in the `sys_prod.xml` to the right one for that region.
* The write_log function is now `log_printf` and works better imo
(the top one was cleaned up by Gary, the bottom one was done entirely by Gary)

### CODE REVIEW

File: source/setup.c [hunk 1]
```diff
@@ -88,6 +103,40 @@ int install_title(int mcp_handle, char *install_dir){
         return ret;
 }
 
+// Both MCP_GetSysProdSettings and MCP_SetSysProdSettings are from recovery_menu and were modified for use with stroopwafel.
```
Comment (jan-hofmeier) [inline source/setup.c]: I would rather have these functions in their own file

File: source/setup.h [hunk 2]
```diff
@@ -2,7 +2,61 @@
 #define SETUP_H
 
 #include <wafel/types.h>
+#include <wafel/ios/svc.h>
```
Comment (jan-hofmeier) [inline source/setup.h]: better have that in it's own header compaining the c file for the Sysprod functions

**Review by jan-hofmeier — CHANGES_REQUESTED** [review]
Move Sysprod stuff to it's own file

File: source/setup.c [hunk 3]
```diff
@@ -216,30 +233,67 @@ u32 setup_main(void* arg){
...
+    int dirHandle = -1;
+    if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040000", &dirHandle) == 0){
+        region = MCP_REGION_JAPAN;
+    } else if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040100", &dirHandle) == 0){
+        region = MCP_REGION_USA;
+    } else if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040200", &dirHandle) == 0){
+        region = MCP_REGION_EUROPE;
+    } else {
+        // handle failure
+        return -1;
```
Comment (jan-hofmeier) [inline source/setup.c]: don't crash the whole setup, just give a warning by changing the LED, write it to log and don't do the sysprod stuff

**Review by jan-hofmeier — CHANGES_REQUESTED** [review]

File: source/setup.h [hunk 4]
```diff
@@ -2,7 +2,22 @@
...
+
+    memset(ptr, 0x00, size);
+
+    return ptr;
+}
+
+static void freeIobuf(void* ptr)
+{
+	iosFree(0xCAFF, ptr);
+}
```
Comment (jan-hofmeier) [inline source/setup.h]: why have this in setup.h?

**Review by jan-hofmeier — CHANGES_REQUESTED** [review]

File: source/setup.h (hunk 4 above)
Comment (Lazr1026) [reply source/setup.h]: should it be in the setup.c?

Comment (Lazr1026) [reply source/setup.h]: or maybe the sysprod.c makes more sense

**Review by jan-hofmeier — APPROVED** [review]

**Pull request merged**